src/library_async.js

```diff
@@ -247,7 +250,13 @@ mergeInto(LibraryManager.library, {
           if (typeof Browser !== 'undefined' && Browser.mainLoop.func) {
             Browser.mainLoop.resume();
           }
-          var asyncWasmReturnValue = Asyncify.doRewind(Asyncify.currData);
+          var asyncWasmReturnValue, isError = false;
+          try {
+            asyncWasmReturnValue = Asyncify.doRewind(Asyncify.currData);
```

Comment:
This worries me, as it seems like any exception happening here would be caught and provided as a Promise rejection? For example, if the rewind process itself hits an error. Ideally this would be focused on the actual execution of code.

Or am I looking at this wrong? I admit I have little experience with handling Promise rejections "properly"...

Reply:
> This worries me, as it seems like any exception happening here would be caught and provided as a Promise rejection?

Well yes, but otherwise the error would be thrown into "nowhere" where it can't be caught. We could limit this to errors from the C/C++ code itself, but I'm not sure if it's a better result - even if it's rewind itself that for some reason errored out, it seems useful to throw that error as part of the original call.

Reply:
What are the cases where rewind itself would fail though? I think I've only ever seen errors from unwind (due to OOM of the Asyncify stack size), but never during rewind...

Reply:
Yes, the real danger us during unwind, not rewind, that's true. Maybe we don't even have assertions there.